libpromises/evalfunction.c

```diff
@@ -2666,14 +2666,23 @@ static FnCallResult FnCallMergeData(EvalContext *ctx, ARG_UNUSED const Policy *p
     assert(false);
 }
 
-JsonElement *DefaultTemplateData(const EvalContext *ctx)
+JsonElement *DefaultTemplateData(const EvalContext *ctx, const char *wantbundle)
 {
     JsonElement *hash = JsonObjectCreate(30);
-    JsonElement *classes = JsonObjectCreate(50);
-    JsonElement *bundles = JsonObjectCreate(50);
-    JsonObjectAppendObject(hash, "classes", classes);
-    JsonObjectAppendObject(hash, "vars", bundles);
+    JsonElement *classes = NULL;
+    JsonElement *bundles = NULL;
 
+    bool want_all_bundles = (NULL == wantbundle);
+
+    if (want_all_bundles) // no specific bundle
+    {
+        classes = JsonObjectCreate(50);
+        bundles = JsonObjectCreate(50);
+        JsonObjectAppendObject(hash, "classes", classes);
+        JsonObjectAppendObject(hash, "vars", bundles);
+    }
+
+    if (want_all_bundles) // no specific bundle
```

Comment:
All same condition. The three ``if (want_all_bundles)`` can be consolidated into one (there is one more below as well).